dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -57,23 +53,88 @@ public CultureInfo Culture
     /// </summary>
     public ILoggerFactory LoggerFactory { get; }
 
+    /// <summary>
+    /// Kernel context reference
+    /// </summary>
+    public IKernel Kernel => this.GetKernelContext();
+
+    /// <summary>
+    /// Spawns the kernel for the context.
+    /// </summary>
+    /// <remarks>
+    /// The kernel context is a lightweight instance of the main kernel with its services.
+    /// </remarks>
+    /// <returns>Kernel reference</returns>
+    private IKernel GetKernelContext()
+        => this._originalKernel; // TODO: Clone a lightweight kernel instead of returning the same instance
+
     /// <summary>
     /// Constructor for the context.
     /// </summary>
+    /// <param name="kernel">Kernel reference</param>
     /// <param name="variables">Context variables to include in context.</param>
     /// <param name="skills">Skills to include in context.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
     public SKContext(
+        IKernel kernel,
         ContextVariables? variables = null,
         IReadOnlySkillCollection? skills = null,
```

Comment:
Should we eliminate the ability to do this and also use the skill collection from the kernel?

Reply:
@markwallace-microsoft At first I considered it, but I didn't want to dictate the skills to be of the Kernel only as we had this interface expecting the parameter already there.